### PR TITLE
Update AngularFire v7 & Firebase v9

### LINKED ISSUE #205: Firebase 9 support
Firebase 9 is currently and beta and support tree shaking
Angularfire 7 (in beta) support Firebase 9 

it'd be lovely to be able to us akita-ng-fire and Firebase with tree shaking (even in beta, and most of the job would be done for the official release)

Thanks

### PR DESCRIPTION
**Pull request by randallmeeker**
resolves #205

### CODE REVIEW

**Comment by randallmeeker** [comment]
Is there a prettier setup that should be used? I'm attempting to add rules that will limit all the white space changes to correspond with the formatting I see already present in the repo.

**Comment by randallmeeker** [comment]
@GrandSchtroumpf sorry mean to tag earlier.

**Comment by GrandSchtroumpf** [comment]
There is no prettier setup because I never managed to achieve something versatile enough for HTML, CSS & TS with prettier. I consider it ~impossible~ very difficult to enforce code understandability with the line width as main parameter. For example [this update](https://github.com/dappsnation/akita-ng-fire/pull/217/files#diff-daa072188c090d5579d7f0fa56a7f54788c484c63fb1a7486199cc5459afa857R69) or [this one](https://github.com/dappsnation/akita-ng-fire/pull/217/files#diff-b52c7395f4d0a58bf5962c0e5f6a4f66c3bad16286044cdadabd926dd48e6fbeR30) look less understandable for me than the previous ones.
I've no issue with adding prettier setup, but I would recommend to not mix PR purpose 🙂.

**Comment by randallmeeker** [comment]
I get that, I was gonna suggest a pre-PR where I just run whatever the Prettier defaults are already in the repo across the entire thing, then this merge into this.

**Comment by randallmeeker** [comment]
@GrandSchtroumpf 

This builds and runs. However, the test suite does not seem to run, but this was the case before I started. Let me know if there are any suggestions there.

I did this off the 6.0.4 brach and as there were changes since v6 that needed updates.

Aside from getting the test suite up and running, I was going to try to upgrade one of my current projects with it, but it appears good to go.

File: package.json [hunk 1]
```diff
@@ -31,7 +31,8 @@
     "@datorama/akita": "^6.1.3",
     "@datorama/akita-ng-router-store": "^6.0.0",
     "akita-ng-fire": "^6.0.0",
-    "firebase": "^8.0.1",
+    "firebase": "^9.1.0",
+    "rxfire": "^6.0.2",
```
Comment (GrandSchtroumpf) [inline package.json:35]: is rxfire a perrDependancy of @angular/fire ?

File: tsconfig.base.json [hunk 2]
```diff
@@ -20,8 +20,5 @@
       ],
       "akita-ng-fire/*": ["dist/akita-ng-fire/*"]
     }
-  },
-  "angularCompilerOptions": {
```
Comment (GrandSchtroumpf) [inline tsconfig.base.json:24]: Is @angular/fire dependent of Ivy ? As far as I remember, it's still consider experimental to build a lib with Ivy

**Comment by GrandSchtroumpf** [comment]
@randallmeeker I've updated the lib to fix this issue #216. I think you'll need to rebase your branch on v6 to have the latest version.
Also I think this is a major update, therefore we should bump the version to 7.0.0.
What we can do for the testing the lib would be to publish this with a next tag. To achieve that, you can 
- bump the version of package.json to `7.0.0-beta.1`
- create a PR against the [branch v7](https://github.com/dappsnation/akita-ng-fire/tree/v7)
- once it's merged, I'll publish version with the tag `next`.
- from there you can update your projects and do the required checks.

File: tsconfig.base.json (hunk 2 above)
Comment (randallmeeker) [reply tsconfig.base.json:24]: ivy is forced enabled in angular 12, this line was telling us it was turning it off, but in reality its now always on. It was flashing up warnings that we were attempting to turn it off so I simply removed this.

File: package.json (hunk 1 above)
Comment (randallmeeker) [reply package.json:35]: It is. `ng add @angular/fire` automatically add it as will `ng update @angular/fire`.

**Comment by GrandSchtroumpf** [comment]
outch I think something went wrong with the rebase here. The commits are in double and you've got conflicts.
I would recommend a new branch fresh from `v7` and cherry pick your commits. I think it would be faster than trying to clean the history / conflicts.

**Pull request closed without merging**